libsolidity/analysis/SyntaxChecker.cpp

```diff
@@ -255,3 +255,10 @@ bool SyntaxChecker::visit(VariableDeclaration const& _declaration)
 	}
 	return true;
 }
+
+bool SyntaxChecker::visit(StructDefinition const& _struct)
+{
+	if (_struct.members().empty())
+		m_errorReporter.syntaxError(_struct.location(), "Defining empty structs is disallowed.");
```

Comment:
This is a breaking change. It should not be too common, but on the other hand, it would be easy to check the pragma here.

Reply:
Perhaps also different error message: `Structs cannot be empty`?

Reply:
I was wondering if it is a breaking change or a bugfix. If it is a non-bugfix then I guess the message should include the "deprecated" word.

Reply:
I'm not sure about this, but since it is easy to make it an experimental 0.5.0 check, I would do it like that.

Reply:
Added. Not entirely convinced with it though. Is the empty struct properly handled in all the ABI encoders?